README.md

```diff
@@ -71,9 +83,341 @@ page preventing our router from loading a view unless the `user` in our `Context
+
+* Render our `UserContext.Provider` inside of our `UserProvider`
+
+* Import our `auth` instance from `services/firebase.js`, listen for changes, and update state accordingly
+
+### Create  `UserProvdier` component
+ 
+UserProvider.js
+```js
+export const UserProvider = (props) => {
+  const [user, setUser] = useState(null);
+   return ()
+  }
+ 
+  
+```
+### Render our new `UserContext.Provider` component 
+```js
+export const UserProvider = (props) => {
+  const [user, setUser] = useState(null);
+   return (
+     <UserContext.Provider value={user}>
+      // the props.children will render any value that is passed to our component without us specifically invoking those             // values in our tempalte
+         <div>{props.children}</div>
+    </UserContext.Provider>
+    )
+  }
+```
+ <strong>Note:</strong> Wait - what is this ` { props.children } ` you may be asking?  Smply put - the `props.children` method is available to us on all components.  It's best to think of it as a placeholder for values <em> we don't know yet </em> when desigining components.  In this case, our `UserProvider` component is rendering our `UserContext.Provider` component which we recieved from calling `createContext()`.  This `UserContext.Provder` component renders `{ props.children }` so that React knows that whatever is nested inside `UserContext.Provider` is rendered in our application.  
+ 
+More on `props.children` [here](https://reactjs.org/docs/composition-vs-inheritance.html) - or try it on [codepen](https://codepen.io/gaearon/pen/ozqNOV?editors=0010)
+
+# Auth and State
+
+Next, lets set up a `useEffect()` in `UserProvider` component to listen for changes in the `auth` object we brought from `firebase.js` by invoking the `onAuthStateChanged()` method from our `auth` object.
+```js
+export const UserProvider = (props) => {
+  const [user, setUser] = useState(null);
+  useEffect(() => {
+//listen for changes
+    auth.onAuthStateChanged(async (user) => {
+		console.log(user)
+      if (user) {
+        // get return values from Firebase
+        const { email, displayName, photoURL, uid } = user;
+      // save them in state
+        setUser({ email, displayName, photoURL, uid });
+      } else {
+        setUser(null);
+      }
+    });
+  }, []);
+  return (
+   // render context provider with user state in value
+    <UserContext.Provider value={user}>
+      <div>{props.children}</div>
+    </UserContext.Provider>
+  );
+};
+```
+<strong>Note:</strong> the `onAuthStateChanged()` method creates something called an `Observer` from a library called RXJS.  This `Observer` listens for any changes on the object it was called on will fire whatever callback we specify once the `Observer` signals a change.  We don't need to get deep under the hood for this lesson but for more on Observers click  [here](https://rxjs.dev/guide/observer)
+
+In order to access our Context - we need to import in in `App.js` and nest our other components inside of it
+
+App.js
```

Comment:
I think, after initial configuration, this is the better place to start. 

[I do like thinking in react](https://reactjs.org/docs/thinking-in-react.html), and building the static/non functional version first, and then adding in functionality. This will allow them to work from the top down:  App => User Provider => UserContext.

The way I work is first I build whatever I am trying to build, then I slim it down, then I try to rebuild the app again in an order that tells a 'story'/would be the way I approach it, if I had a second chance to build with more order and clarity. 

I know the clock is ticking, so no worries if there is not time